sdk/core/core-lro/test/lro.spec.ts

```diff
@@ -2255,15 +2255,16 @@ matrix(
             },
           });
           assert.equal(serialized, expectedSerialized);
-          assert.equal(poller.operationState.status, "succeeded");
+          assert.isNotNull(poller.operationState);
+          assert.equal(poller.operationState!.status, "succeeded");
```

Comment:
@jeremymeng Is it possible we could report this error during `dev-tool run test:vitest ...`?

Reply:
It reports them as warnings already just not failing the tests.  We could potentially have some additional step to type-check tests and samples-dev code. I will think about this.

@maorleger @mpodwysocki I remember there's some vitest assertion that could allow us to get rid of the `!`? `assert.exists` or something similar?